src/go/rpk/pkg/cli/cmd/container/start.go

```diff
@@ -114,12 +168,109 @@ func newStartCommand() *cobra.Command {
 	)
 	command.Flags().MarkHidden(imageFlag)
 
+	command.Flags().StringSliceVar(
+		&kafkaPorts,
+		"kafka-ports",
+		[]string{strconv.Itoa(config.DefaultKafkaPort)},
+		"List of ports that you want to use. Ports must be surely free otherwise containers won't start",
+	)
+
+	// TODO: fill in the usage strings
+	command.Flags().StringSliceVar(
+		&proxyPorts,
+		"proxy-ports",
+		[]string{"7072"},
+		"",
+	)
+	command.Flags().StringSliceVar(
+		&registryPorts,
+		"registry-ports",
+		[]string{strconv.Itoa(config.DefaultSchemaRegPort)},
+		"",
+	)
+	command.Flags().StringSliceVar(
+		&adminPorts,
+		"admin-ports",
+		[]string{strconv.Itoa(config.DefaultAdminPort)},
+		"",
+	)
+	command.Flags().StringSliceVar(
+		&brokerRPCPorts,
+		"broker-rpc-ports",
+		[]string{"33145"},
+		"",
+	)
+
 	return command
 }
 
+func assignPorts(definedPorts []string, nodesCount uint, portName string) ([]uint, error) {
+	if len(definedPorts) > 0 && len(definedPorts) > int(nodesCount) {
+		return nil, fmt.Errorf("--%s should not exceed the total number of nodes (%d)", portName, nodesCount)
+	}
+	if len(definedPorts) == 0 {
+		return nil, nil
+	}
+	ports := make([]uint, 0, len(definedPorts))
+	var any, none, withPort bool
+
+	for _, definedPort := range definedPorts {
+		switch definedPort {
+		case "none":
+			none = true
+		case "any":
+			any = true
+		default:
+			parsedPort, err := strconv.ParseUint(definedPort, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("provided port %q is not a valid uint nor a valid string ('any', 'none')", definedPort)
+			}
+			if parsedPort > 65535 {
+				return nil, fmt.Errorf("provided port %q exceeded the range of transport layer range of 0 - 65535", definedPort)
+			}
+			err = vnet.CheckIfPortIsFree(uint(parsedPort))
+			if err != nil {
+				return nil, fmt.Errorf("specified port %q is not available", definedPort)
+			}
+			withPort = true
+		}
+	}
+	if any && none {
+		return nil, fmt.Errorf("cannot specify 'any' and 'none' together")
+	}
+	if (withPort && any) || (withPort && none) {
+		return nil, fmt.Errorf("cannot specify 'any' or 'none' with specific ports")
+	}
+
+	switch {
+	case any:
+		var err error
+		ports, err = vnet.GetFreePortPool(int(nodesCount))
```

Comment:
`assignPorts` is currently is called one flag at a time -- with the pool, we need to allocate _all_ ports up front. The reason is that, as the code currently is, we may allocate ports in one `assignPorts` call, and then allocate those same ports in the next `assignPorts` call.

Perhaps to make it easy, you can allocate the worst case up front: count how many you'll need to allocate, reserve that pool, and then pass the pool to this function for every flag.